dart/dynamics/detail/BodyNode.hpp

```diff
@@ -153,6 +153,23 @@ ShapeNode* BodyNode::createShapeNode(ShapeNodeProperties properties,
   return createNode<ShapeNode>(properties);
 }
 
+//==============================================================================
+template <class ShapeType>
+ShapeNode* BodyNode::createShapeNode(const std::shared_ptr<ShapeType>& shape)
+{
+  return createShapeNode(static_cast<ShapePtr>(shape));
```

Comment:
Could we remove the non-template function by having the implementation here? I think this template function can handle the case as well. :thinking: If this works, then we could apply this to the below function as well.

Reply:
Good point. I'll try that out right now and then push it if it works (which I fully expect it will).

Reply:
The change worked, so I've pushed it.

The one caveat with this is it breaks ABI compatibility, so anyone who was using that function will need to recompile. We don't support ABI compatibility between minor versions, so this doesn't violate our guidelines, but it's something to watch out for.

Reply:
On the bright side, we've reduced three functions to two while also expanding their usability.

Reply:
Yeah, I like this change. Btw, I recall that the same issue was of `createShapeNode(shape, name)`. I couldn't `const char*` because of the template function. Now everything is resolved in a neat way!